src/libs/workspace-utils.ts

```diff
@@ -188,10 +198,10 @@ export interface WorkspaceAccessInfo {
 export const canEditWorkspace = ({
   accessLevel,
   workspace: { isLocked },
-}: WorkspaceAccessInfo): { value: boolean; message?: string } =>
+}: WorkspaceAccessInfo | WorkspaceWrapper): { value: boolean; message?: string } =>
   cond<{ value: boolean; message?: string }>(
     [!canWrite(accessLevel), () => ({ value: false, message: 'You do not have permission to modify this workspace.' })],
-    [isLocked, () => ({ value: false, message: 'This workspace is locked.' })],
+    [!!isLocked, () => ({ value: false, message: 'This workspace is locked.' })],
```

Comment:
Does this need to be coerced? Isn't `isLocked` already a boolean?

Reply:
It does.... isLocked is currently an optional value on a workspace.
Not coercing it here just moves where this is happening (and forces the de-and-re-construction of objects in a bunch of places).
I really need to go through and clean up all the `useWorkspace` hooks so they're returning types that enumerate the included fields.

Reply:
I'm not keen on this direction...

What we want to get to is where functions that use workspaces are typed with the fields they include (thus, `isLocked` would not be optional here). Then `useWorkspaces` and other data fetching functions type the workspace objects they return and we use TypeScript to verify that we're loading the fields we need.

Reply:
I'm fine with using `@ts-expect-error` for now